experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -358,9 +369,12 @@ static int ws_eink_update_display(struct ws_eink_fb_par *par)
 	if (ret)
 		return ret;
 
-	memcpy(&ssbuf, &vmem, sizeof(vmem));
+	frame_size = par->props->height * par->props->width * par->props->bpp / 8;
 
-	ret = set_frame_memory(par, ssbuf);
+	for (i = 0; i < frame_size; i++) {
```

Comment:
I think this won't work in the general case.  There is no guarantee that the entire `vmem` has been rewritten each time `ws_eink_update_display` is called.  I think what you need to do is create a local copy of `vmem` and call `bitrev8` on each byte of that copy.  Then pass that copy to `set_frame_memory`.  Once `set_frame_memory` is done, the temporary bit reversed copy of `vmem` can be discarded.